File: Examples/CSharp/Working-with-SolutionXML-Elements/AddSolutionXMLElement.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_SolutionXML_Elements
{
    public class AddSolutionXMLElement
    {
        public static void Run() 
        {
            // ExStart:AddSolutionXMLElement
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_SolutionXML();

            // Load source Visio diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");

            // Initialize SolutionXML object
            SolutionXML solXML = new SolutionXML();
            // Set name
            solXML.Name = "Solution XML";
            // Set xml value
            solXML.XmlValue = "XML Value";
            // Add SolutionXML element
            diagram.SolutionXMLs.Add(solXML);

            // Save Visio diagram
            diagram.Save(dataDir + "AddSolutionXMLElement_Out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:AddSolutionXMLElement
        }
    }
}

File: Examples/VisualBasic/Working-with-SolutionXML-Elements/AddSolutionXMLElement.vb
Imports Aspose.Diagram
Imports System
Public Class AddSolutionXMLElement
    Public Shared Sub Run()
        ' ExStart:AddSolutionXMLElement
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_SolutionXML()

        ' Load source Visio diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))

        ' Initialize SolutionXML object
        Dim solXML As New SolutionXML()
        ' Set name
        solXML.Name = "Solution XML"
        ' Set xml value
        solXML.XmlValue = "XML Value"
        ' Add SolutionXML element
        diagram.SolutionXMLs.Add(solXML)

        ' Save Visio diagram
        diagram.Save(dataDir & Convert.ToString("AddSolutionXMLElement_out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:AddSolutionXMLElement
    End Sub
End Class
